doc/nrf/security/tfm.rst

```diff
@@ -253,38 +293,461 @@ We will decrease the size of the (optional) ``mcuboot_pad`` partition and thus t
+The random seed is generated once in the boot process and stays the same until reset.
+
+.. _tfm_partition_its_sizing:
+
+Sizing the Internal Trusted Storage
+-----------------------------------
+
+The RAM and flash usage of the ITS service are included in the ``tfm_secure`` partition.
+The storage itself is a separate ``tfm_its`` partition.
+
+When using the :ref:`partition_manager`, you can configure the size of the ``tfm_its`` with the :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM_INTERNAL_TRUSTED_STORAGE` Kconfig option.
+The resulting partition is visible in the :file:`partitions.yml` file in the build directory:
+
+.. code-block:: console
+
+    EMPTY_2:
+      address: 0xea000
+      end_address: 0xf0000
+      placement:
+        after:
+        - tfm_its
+      region: flash_primary
+      size: 0x6000
+    tfm_its:
+      address: 0xe8000
+      end_address: 0xea000
+      inside:
+      - tfm_storage
+      placement:
+        align:
+          start: 0x8000
+        before:
+        - tfm_otp_nv_counters
+      region: flash_primary
+      size: 0x2000
+
+The :ref:`partition_manager` can only align the start address of the ``tfm_its`` partition with the flash region size (see :ref:`ug_tfm_partition_alignment_requirements`).
+If the size of the ``tfm_its`` does not equal the flash region size, the Partition Manager allocates an additional empty partition to fill the gap.
+See the :ref:`tfm_ps_static_partition` for an example on how to optimize the size of the ``tfm_its`` partition by manual configuration.
+
+TF-M does not guarantee in build time that the ``tfm_its`` partition can hold the assets that are configured with the :kconfig:option:`CONFIG_TFM_ITS_NUM_ASSETS` and :kconfig:option:`CONFIG_TFM_ITS_MAX_ASSET_SIZE` options.
+Depending on the available flash size, the ITS can use one or two flash pages (4 KB) for ensuring power failure safe operations.
+In addition, ITS stores the bookkeeping information for the assets in the flash memory and the bookkeeping size scales with the configured number of assets.
+This can leave a very small amount of space for the actual assets.
+
+It is recommended to test the ITS with the intended assets to ensure that the assets fit in the available space.
+
+.. _tfm_partition_crypto:
+
+Crypto service
+==============
+
+The crypto service implements the PSA Crypto APIs (`PSA Certified Crypto API`_) and provides cryptographic services to other TF-M partitions and to the non-secure application.
+
+To enable the crypto service, set the :kconfig:option:`CONFIG_TFM_PARTITION_CRYPTO` Kconfig option.
+
+You can configure the service directly using the ``CONFIG_TFM_CRYPTO_*`` Kconfig options found in the :file:`zephyr/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules` file.
+However, it is recommended to use the ``CONFIG_PSA_WANT_*`` Kconfig options to enable the required algorithms and key types.
+These will enable the required ``CONFIG_TFM_CRYPTO_*`` Kconfig options.
```

Comment:
Just curious, is there a reason these don't have the :kconfig:option: markup too?

Reply:
Never mind, Pekka told me.